src/docs/development/add-to-app/ios/add-flutter-screen.md

```diff
@@ -6,77 +6,291 @@ description: Learn how to add a single Flutter screen to your existing iOS app.
+demonstrate creating a `FlutterEngine`, exposed as a property, on app startup in
+the app delegate.
+
+<ul class="nav nav-tabs" id="engine-language" role="tablist">
+  <li class="nav-item">
+    <a class="nav-link active" id="engine-objc" href="#engine-objc-tab" role="tab" aria-controls="engine-objc" aria-selected="true">Objective-C</a>
+  </li>
+  <li class="nav-item">
+    <a class="nav-link" id="engine-swift" href="#engine-swift-tab" role="tab" aria-controls="engine-swift" aria-selected="false">Swift</a>
+  </li>
+</ul>
+
+<div class="tab-content"> {% comment %} FlutterEngine language tab start {% endcomment -%}
+
+<div class="tab-pane active" id="engine-objc-tab" role="tabpanel" aria-labelledby="engine-objc-tab" markdown="1">
+
+**In `AppDelegate.h`:**
+
+<?code-excerpt "AppDelegate.h" title?>
+```objectivec
 @import UIKit;
 @import Flutter;
 
-@interface AppDelegate : FlutterAppDelegate
+@interface AppDelegate : FlutterAppDelegate // More on the FlutterAppDelegate below.
 @property (nonatomic,strong) FlutterEngine *flutterEngine;
 @end
 ```
 
-This allows `AppDelegate.m` to be simple, unless your host app needs to override
-other methods here:
+**In `AppDelegate.m`:**
 
-```objective-c
-#import <FlutterPluginRegistrant/GeneratedPluginRegistrant.h> // Only if you have Flutter Plugins
+<?code-excerpt "AppDelegate.m" title?>
+```objectivec
+#import <FlutterPluginRegistrant/GeneratedPluginRegistrant.h> // Used to connect plugins.
 
 #import "AppDelegate.h"
 
 @implementation AppDelegate
 
-// This override can be omitted if you do not have any Flutter Plugins.
 - (BOOL)application:(UIApplication *)application
     didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
-  self.flutterEngine = [[FlutterEngine alloc] initWithName:@"io.flutter" project:nil];
-  [self.flutterEngine runWithEntrypoint:nil];
+  self.flutterEngine = [[FlutterEngine alloc] initWithName:@"my flutter engine"];
+  // Runs the default Dart entrypoint with a default Flutter route.
+  [self.flutterEngine run];
   [GeneratedPluginRegistrant registerWithRegistry:self.flutterEngine];
   return [super application:application didFinishLaunchingWithOptions:launchOptions];
 }
 
 @end
 ```
 
-If you are writing in Swift, you can do the following in your `AppDelegate.swift`:
+</div>
+
+<div class="tab-pane" id="engine-swift-tab" role="tabpanel" aria-labelledby="engine-swift-tab" markdown="1">
 
+**In `AppDelegate.swift`:**
+
+<?code-excerpt "AppDelegate.swift" title?>
 ```swift
 import UIKit
 import Flutter
-import FlutterPluginRegistrant // Only if you have Flutter Plugins.
+import FlutterPluginRegistrant // Used to connect plugins.
 
 @UIApplicationMain
-class AppDelegate: FlutterAppDelegate {
-  var flutterEngine : FlutterEngine?;
-  // Only if you have Flutter plugins.
+class AppDelegate: FlutterAppDelegate { // More on the FlutterAppDelegate below.
+  lazy var flutterEngine = FlutterEngine(name: "my flutter engine")
```

Comment:
Do we have engine naming conventions?

Reply:
looking into it a bit, this only adds a debug label for the thread in instruments. I want to get rid of it so I'll not over-emphasize it here for now